src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -66,23 +66,34 @@ internal sealed class State
 
             public void RemoveStream(MsQuicStream stream)
             {
+                bool closing;
                 lock (this)
                 {
                     StreamCount--;
+                    closing = _closingGCHandle != IntPtr.Zero && StreamCount == 0;
                 }
 
-                if (Closing && StreamCount == 0)
+                if (closing)
                 {
                     Handle?.Dispose();
-                    //if (_stateHandle.IsAllocated) _stateHandle.Free();
+                    try
+                    {
+                        GCHandle gcHandle = GCHandle.FromIntPtr(_closingGCHandle);
+                        Debug.Assert(gcHandle.IsAllocated);
+                        if (gcHandle.IsAllocated) gcHandle.Free();
+                    }
+                    catch (Exception ex)
```

Comment:
I do not think you should be catching the exception here. Freeing GCHandle is not supposed to throw, and when it does throw, it is likely a bad GC heap corruption and it would be better to leave the exception unhandled and crash the process in that case.

Reply:
ok. note that this is likely invoked from native callback. But I could see that getting core at this point may be useful.

Reply:
updated